internal/extractors/aez_reader.go

```diff
@@ -17,28 +19,117 @@ type PackageType interface {
 // DecodeAEZ processes data packages
 func DecodeAEZ(target chan<- common.DataRecord, source <-chan common.DataRecord) {
 	defer close(target)
+	var multiPackBuffer *bytes.Buffer
+
 	for sourcePacket := range source {
 		if sourcePacket.Error != nil {
 			target <- sourcePacket
 			continue
 		}
 		reader := bytes.NewReader(sourcePacket.Buffer)
-		if sourcePacket.SourceHeader.PacketSequenceControl.GroupingFlags() == innosat.SPStandalone {
-			switch {
-			case sourcePacket.TMHeader.IsHousekeeping():
-				var sid aez.SID
-				binary.Read(reader, binary.BigEndian, &sid)
-				targetPacket, err := instrumentHK(sid, reader)
-				sourcePacket.Data = targetPacket
-				sourcePacket.SID = sid
+		switch sourcePacket.SourceHeader.PacketSequenceControl.GroupingFlags() {
+		case innosat.SPStandalone:
+			// Panic if has started multipack
+			if multiPackBuffer != nil {
+				target <- prepareUnfinishedMultiPackPanic(multiPackBuffer, sourcePacket)
+				multiPackBuffer = nil
+			}
+			target <- processSourcePacket(reader, sourcePacket)
+		case innosat.SPStart:
+			// Panic if hasn't concluded previous
+			if multiPackBuffer != nil {
+				target <- prepareUnfinishedMultiPackPanic(multiPackBuffer, sourcePacket)
```

Comment:
You could use another channel as a queue, i.e. sending a part of a jpeg ([]byte) to a go routine that simply appends every chunk it gets to a larger byte array. Then the Exctract data could wait for the "result" if it recieves a "start header" package - giving the extract data function the resposibility to asseble the information header + data.

Reply:
if we argue for that, I'd say this outer tier should not be an aez_reader at all but only be a package sorter so we have

```
sorter -> multipack aggregator
 |                      |
 \-------------------------------------> eaz_reader
```

but I don't really see how it does anything but adds complexity.

I really don't like the idea of having the first tier peaking into the packages to see what they are.

Reply:
Or well a split of the `sorter` and the `aez_reader` would make sense though. I'll do that.

Reply:
I consider the aggregator I've now added to resolve this.